File: music-module/src/main/java/per/zsck/music/service/MusicService.java
package per.zsck.music.service;


import org.bson.types.ObjectId;
import org.springframework.data.mongodb.gridfs.GridFsResource;
import per.zsck.music.entity.Music;

import java.util.List;

/**
 * @author zsck
 * @date 2022/12/11 - 11:35
 */

public interface MusicService {

    boolean removeMusicByFileId(String fileId);

    GridFsResource getMusicFileByFileId(String fileId);

    List<Music> likeMusic(String keyword);

    ObjectId uploadMusicAndReplace(Music music, byte[] bytes);

    Music analysisAndUpload(String fileName, byte[] bytes);

}

File: src/main/kotlin/per/zsck/simbot/http/kugou/service/MusicService.kt
package per.zsck.simbot.http.kugou.service

import org.springframework.data.mongodb.core.MongoTemplate
import org.springframework.data.mongodb.core.query.Criteria
import org.springframework.data.mongodb.core.query.Query
import org.springframework.stereotype.Service
import per.zsck.simbot.http.kugou.entity.Music
import java.util.regex.Pattern
import javax.annotation.Resource

/**
 * @author zsck
 * @date   2022/11/10 - 13:04
 */
interface MusicService {
    fun likeMusic(audioName: String): MutableList<Music>
}

@Service
class MusicServiceImpl : MusicService {

    @Resource
    lateinit var mongoTemplate: MongoTemplate

    override fun likeMusic(audioName: String): MutableList<Music> {
        return mongoTemplate.find( Query.query(Criteria.where( "audioNameIndex" )
            .regex( Pattern.compile(".*${audioName}.*", Pattern.CASE_INSENSITIVE) )), Music::class.java )
    }

}